src/hooks/useExperimentConfig.js

```diff
@@ -0,0 +1,61 @@
+import { useEffect } from 'preact/hooks'
+import { getExperimentConfig } from 'api'
+import { isConsentGranted, setExperimentGroup, clearExperimentGroup } from 'utils'
+
+const STORAGE_KEY = '2021-KaiOS-app-engagement-config'
+const DAY_TIMESTAMP = 24 * 60 * 60 * 1000
+
+function formatDate (date) {
+  var d = new Date(date)
+  var month = '' + (d.getMonth() + 1)
+  var day = '' + d.getDate()
+  var year = d.getFullYear()
+
+  if (month.length < 2) { month = '0' + month }
+  if (day.length < 2) { day = '0' + day }
+
+  return [year, month, day].join('')
+}
+
+export const useExperimentConfig = () => {
+  const consentGranted = isConsentGranted()
+  useEffect(() => {
+    const nowTimestamp = Date.now()
+    const { timestamp } = JSON.parse(localStorage.getItem(STORAGE_KEY)) || {}
+    const hasRecordBefore = timestamp && ((nowTimestamp - timestamp) < DAY_TIMESTAMP)
+
+    if (!hasRecordBefore && consentGranted) {
+      const [promise,, xhr] = getExperimentConfig()
```

Comment:
Please keep `xhr` encapsulated inside cachedFetch. You can extract and parse the cookies in the API layer.